lib/mayaUsd/ufe/ProxyShapeCameraHandler.cpp

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "ProxyShapeCameraHandler.h"
+#include "UsdCameraHandler.h"
+#include "Global.h"
+
+#include "pxr/usd/usdGeom/camera.h"
+
+#include <mayaUsd/ufe/UsdSceneItem.h>
+#include <mayaUsd/ufe/Utils.h>
+
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usd/primRange.h>
+#include <pxr/usd/usd/stage.h>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+ProxyShapeCameraHandler::ProxyShapeCameraHandler(const Ufe::CameraHandler::Ptr& mayaCameraHandler)
+    : Ufe::CameraHandler()
+    , fMayaCameraHandler(mayaCameraHandler)
+{
+}
+
+ProxyShapeCameraHandler::~ProxyShapeCameraHandler() { }
+
+/*static*/
+ProxyShapeCameraHandler::Ptr ProxyShapeCameraHandler::create(const Ufe::CameraHandler::Ptr& mayaCameraHandler) { return std::make_shared<ProxyShapeCameraHandler>(mayaCameraHandler); }
+
+//------------------------------------------------------------------------------
+// Ufe::CameraHandler overrides
+//------------------------------------------------------------------------------
+Ufe::Camera::Ptr ProxyShapeCameraHandler::camera(const Ufe::SceneItem::Ptr& item) const
+{
+    return nullptr;
+}
+
+Ufe::Selection ProxyShapeCameraHandler::find(const Ufe::Path& path) const
+{
+    // Allow the gateway item to also have children in the same runtime.
```

Comment:
Do we really need to do this?  The ProxyShape is a shape node --- shape nodes only have children in limited circumstances, and can they really be cameras?